### PR TITLE
Introduce Custom Prelude for Improved Code Organization and Convenience

### PR DESCRIPTION
**Pull request by malik672**
Motivation:
issue #16 
Currently, our codebase has scattered import statements across modules, leading to a lack of consistency and increased verbosity. This pull request introduces a custom prelude to centralize frequently used imports, improving code organization and developer experience.

Changes Made:

The creation of the prelude.rs file.
Re-exported items from various modules.
Any significant changes to existing modules or files.

Impact:

These changes are backward-compatible and should not impact existing functionality. Developers using our crate can seamlessly transition to the new prelude for a more streamlined import experience.

### CODE REVIEW

**Comment by malik672** [comment]
definitely failing the lint check

**Comment by malik672** [comment]
@shuhuiluo what command did you use for the linter, checked the logs and saw cargo fmt, ran it in my terminal and it displayed nothing

File: src/prelude.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+pub use crate::{
+    constants::{Rounding, MAX_UINT256},
+    entities::{
+        base_currency::{BaseCurrency, CurrencyLike},
+        currency::CurrencyTrait,
+        ether::Ether,
+        fractions::{
+            currency_amount::CurrencyAmount,
+            fraction::{Fraction, FractionLike, FractionTrait},
+        },
```
Comment (shuhuiluo) [inline src/prelude.rs:11]: Missing `percent` and `price`.

Comment (malik672) [reply src/prelude.rs:11]: > Missing `percent` and `price`.

don't think they are used anywhere or depends on who make the next pull we can just add it. can I add you as a reviewer.

Comment (shuhuiluo) [reply src/prelude.rs:11]: Sure.

**Pull request merged**